src/GitHub.Exports/Models/DiffUtilities.cs

```diff
@@ -113,6 +135,37 @@ public static DiffLine Match(IEnumerable<DiffChunk> diff, IList<DiffLine> target
             return null;
         }
 
+        static string NormalizeLineEndings(string text)
+        {
+            return text.Replace("\r\n", "\n");
+        }
+
+        static bool HasCarriageReturn(string text)
+        {
+            return text.IndexOf('\r') != -1;
+        }
+
+        static string EscapeCarriageReturns(string text)
+        {
+            return text.Replace("\r", "\\r");
+        }
+
+        static string UnescapeCarriageReturns(string text)
+        {
+            return text.Replace("\\r", "\r");
```

Comment:
What will happen here if the diff already has the string "\\r" in it? It will get replaced by a control character, but will that affect the correctness of the parsing?

Reply:
Yes, indeed. Poorly thought through escaping. This code is redundant now.